clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -665,36 +677,34 @@ void CIRGenFunction::buildStoreThroughBitfieldLValue(RValue Src, LValue Dst,
 
     mlir::Value Val = buildLoadOfScalar(Dst, Dst.getPointer().getLoc());
 
-    mlir::Type Ty = SrcVal.getType();
     // Mask the source value as needed.
     if (!hasBooleanRepresentation(Dst.getType()))
-      SrcVal =
-          builder.createBinop(SrcVal, mlir::cir::BinOpKind::And,
-                              llvm::APInt::getLowBitsSet(SrcWidth, Info.Size));
-
+      SrcVal = builder.createAnd(SrcVal, llvm::APInt::getLowBitsSet(SrcWidth, Info.Size));
+      
     MaskedVal = SrcVal;
     if (Offset)
-      SrcVal = builder.createShift(SrcVal, llvm::APInt(SrcWidth, Offset), true);
+      SrcVal = builder.createShiftLeft(SrcVal, Offset);
 
     // Mask out the original value.
-    Val = builder.createBinop(
-        Val, mlir::cir::BinOpKind::And,
+    Val = builder.createAnd(Val,        
         ~llvm::APInt::getBitsSet(SrcWidth, Offset, Offset + Info.Size));
 
     // Or together the unchanged values and the source value.
-    SrcVal = builder.create<mlir::cir::BinOp>(
-        Val.getLoc(), Ty, mlir::cir::BinOpKind::Or, Val, SrcVal);
+    SrcVal = builder.createOr(Val, SrcVal);
+
   } else {
-    assert(Offset == 0);
+    assert(0 && "not implemented");
```

Comment:
NIT: use llvm_unreachable over assert and use a more descriptive message (e.g. "feature X is NYI")